### PR TITLE
add isFetching key to state.app

### PR DESCRIPTION
**Pull request by mmcgahan**
This will help provide more accurate 'loading' indicators in the UI for

1. page transitions
2. lazy loading of data
3. in-flight POSTs

AND MORE!!

### CODE REVIEW

File: src/reducers/platform.js [hunk 1]
```diff
@@ -24,21 +24,23 @@ export function app(state=DEFAULT_APP_STATE, action={}) {
 
 	switch (action.type) {
 	case 'API_REQUEST':
-		if (action.meta.logout) {
+		if ((action.meta || {}).logout) {
```
Comment (mmcgahan) [inline src/reducers/platform.js]: this is sort of an unrelated change, but prevents an annoying requirement for `apiRequest` calls to always supply a `meta` argument.

File: src/reducers/platform.js [hunk 2]
```diff
@@ -24,21 +24,23 @@ export function app(state=DEFAULT_APP_STATE, action={}) {
...
 			return DEFAULT_APP_STATE;  // clear app state during logout
 		}
-		return state;
-	case 'CACHE_SUCCESS':  // fall through - same effect as API success
+		return { ...state, isFetching: true };
 	case 'API_SUCCESS':
-		// API_SUCCESS contains an array of responses, but we just need to build a single
+		state.isFetching = false;  // fall through - everything else is the same as CACHE_SUCCCESS
```
Comment (mmcgahan) [inline src/reducers/platform.js]: make sure you use the split view in github for this section of code - the 'Unified' version has some weird change groupings

File: src/reducers/platform.js [hunk 3]
```diff
@@ -71,16 +73,9 @@ export function clickTracking(state=DEFAULT_CLICK_TRACK, action) {
...
 
 export function config(state={}, action) {
-	let apiUrl,
-		trackId;
-
 	switch(action.type) {
 	case 'CONFIGURE_API_URL':
-		apiUrl = action.payload;
-		return { ...state, apiUrl };
-	case 'CONFIGURE_TRACKING_ID':
```
Comment (mmcgahan) [inline src/reducers/platform.js]: this is dead code cleanup - tracking ids are managed entirely by cookies, not Redux

File: src/reducers/platform.js [hunk 4]
```diff
@@ -24,21 +24,23 @@ export function app(state=DEFAULT_APP_STATE, action={}) {
...
+		return { ...state, isFetching: true };
 	case 'API_SUCCESS':
-		// API_SUCCESS contains an array of responses, but we just need to build a single
+		state.isFetching = false;  // fall through - everything else is the same as CACHE_SUCCCESS
+	case 'CACHE_SUCCESS':
+		// {API|CACHE}_SUCCESS contains an array of responses, but we just need to build a single
 		// object to update state with
 		newState = action.payload.responses.reduce((s, r) => ({ ...s, ...r }), {});
 		delete state.error;
 		return { ...state, ...newState };
```
Comment (akdetrick) [inline src/reducers/platform.js]: assuming that if `CACHE_SUCCESS` is dispatched, we never make a request?

Comment (akdetrick) [inline src/reducers/platform.js]: nvm, the tests cleared this up

File: src/reducers/platform.test.js [hunk 5]
```diff
@@ -39,6 +40,31 @@ describe('app reducer', () => {
...
+		};
+		const appState = app(undefined, API_SUCCESS);
+		expect(appState.isFetching).toBe(false);
+	});
+	it('does not change isFetching on CACHE_SUCCESS', () => {
+		const CACHE_SUCCESS = {
+			type: 'CACHE_SUCCESS',
+			payload: { queries: [], responses: [] },
+		};
+		const appState = app({ isFetching: true }, CACHE_SUCCESS);
```
Comment (akdetrick) [inline src/reducers/platform.test.js]: Should we also test that `isFetching: false` remains `false` on `CACHE_SUCCESS`?

Comment (mmcgahan) [inline src/reducers/platform.test.js]: probs - will update shortly

File: src/reducers/platform.js (hunk 2 above)
Comment (akdetrick) [inline src/reducers/platform.js]: i'm split view 4 life

**Pull request merged**